### ISSUE TITLE
Bootstrapping an admin user or service account at server startup

### ISSUE THREAD

**Issue by vmuzikar** [keycloak/keycloak#30003]
### Description

Based on the design as specified in #9829.

* Usable only if Keycloak was not started before, i.e. the master realm doesn't not exist.
* When the server is starting (`kc.sh|bat start|start-dev`) and the condition specified above is met, it will check the env vars as specified below.

### Admin user
* The following env vars are checked: `KEYCLOAK_BOOTSTRAP_ADMIN_USERNAME`, `KEYCLOAK_BOOTSTRAP_ADMIN_PASSWORD` and `KEYCLOAK_BOOTSTRAP_ADMIN_EXPIRATION`. Their meaning and functionality is identical to as when used as the [dedicated command](https://github.com/keycloak/keycloak/issues/29324).
* If `KEYCLOAK_BOOTSTRAP_ADMIN_USERNAME` is set but `KEYCLOAK_BOOTSTRAP_ADMIN_PASSWORD` is not, an error will be shown.
* It is possible to set only `KEYCLOAK_BOOTSTRAP_ADMIN_PASSWORD`. In that case `KEYCLOAK_BOOTSTRAP_ADMIN_USERNAME` defaults to `temp-admin`.

### Admin service account
* The following env vars are checked: `KEYCLOAK_BOOTSTRAP_ADMIN_CLIENT_ID`, `KEYCLOAK_BOOTSTRAP_ADMIN_CLIENT_SECRET` and `KEYCLOAK_BOOTSTRAP_ADMIN_EXPIRATION`. Their meaning and functionality is identical to as when used as the [dedicated command](https://github.com/keycloak/keycloak/issues/30002).
* If `KEYCLOAK_BOOTSTRAP_ADMIN_CLIENT_ID` is set but `KEYCLOAK_BOOTSTRAP_ADMIN_CLIENT_SECRET` is not, an error will be shown.
* It is possible to set only `KEYCLOAK_BOOTSTRAP_ADMIN_CLIENT_SECRET`. In that case `KEYCLOAK_BOOTSTRAP_ADMIN_USERNAME` defaults to `temp-admin-service`.

**Comment by vmuzikar** [comment]
As per @stianst's suggestion, the startup (`kc.sh|bat start|start-dev`) in addition to env vars should also accept CLI options (similarly to the dedicated `kc.sh|bat admin-bootstrap` command). Something like `--bootstrap-admin-username`, `--bootstrap-admin-password`, `--bootstrap-admin-client-id`, `--bootstrap-admin-client-secret`. I think it makes sense for use cases where env vars are not desirable (e.g. when running tests).

WDYT @keycloak/cloud-native?

**Comment by shawkins** [comment]
> I think it makes sense for use cases where env vars are not desirable (e.g. when running tests)

Would these be full kc.bootstrap-admin-* / hidden options? If so they will end up being a little more powerful than the options on the bootstrap-admin commands in the draft pr as those aren't currently using the small rye configuration.

Or the existing initial admin logic is looking at system properties in addition to the env variables - we can continue to do that here to skip the option / property mapper logic.

**Comment by vmuzikar** [comment]
Just to summarize our offline discussion, we aligned on having full blown Keycloak options (SmallRye Config options) for the start|start-dev command:
```
kc.sh start --bootstrap-admin-user --bootstrap-admin-password --bootstrap-admin-client-id --bootstrap-admin-client-secret
```
The options will be marked as hidden.

**Comment by vmuzikar** [comment]
Resolved by #30054.
